src/TopPage/DataExtension.php

```diff
@@ -205,13 +231,47 @@ public function withTopPageUpdate(bool $update, callable $callback)
         }
     }
 
+    /**
+     * Use this to wrap any code which is supposed to run with fixed top page
+     * Useful when top page is known upfront and doesn't need to be determined
+     * For example: model duplication where parent is assigned and saved only after the duplication is done
+     * It's not possible to determine top page in such case however it might be possible to know the top page
+     * even before the operation starts from the specific context
+     * Setting the page id to null disables this feature
+     *
+     * @param int|null $topPageId
+     * @param callable $callback
+     * @return mixed
+     */
+    public function withFixedTopPage(?int $topPageId, callable $callback)
```

Comment:
Very odd having `?int` for the first arg.  In what scenario would you call `$myobj->withFixedTopPage(null, function ...)`?

Reply:
`null` is a valid value to pass to this feature which effectively disables it. I agree that this is probably a marginal case but it may come in handy for example in unit tests.